toplevel/toploop.mli

```diff
@@ -15,6 +15,12 @@
 
 open Format
 
+(* type of toplevel usages *)
+type usage_types =
+  | File
+  | Stdin
+  | String
+type usage = {value: string; tag: usage_types}
```

Comment:
I don't understand the API design here: I'm not sure what `tag` means, and why you are using the name `usage`. It looks like what is being described is the source/input of the code to execute?

Also, making this a product type (a record here) is not so nice as some of the options are not meaningful: as far as I understand `{tag = Stdin; value = "foo"}` will always ignore the `value` part. Could this be a sum type, `| File of string | Stdin | String of string`?

Reply:
moved this stuff to sum types